src/util/entitlements/helpers.tsx

```diff
@@ -0,0 +1,10 @@
+export const hasEntitlement = (
+  required: string,
+  entitlements?: string[],
+  isFineGrained: boolean | null = false,
+): boolean => {
+  if (!isFineGrained) {
```

Comment:
```suggestion
  if (isFineGrained === false) {
```
I think we need to be specific here. Actually I am not sure what the right thing to do is in case isFineGrained is null.

Reply:
if `isFineGrained` is `null`, that means the api call to get identity status did not yet complete. In that case we should not grant user access to restricted actions in the UI. The purpose of the `/1.0/identities/current` endpoint is that an identity should always be able to confirm its own "restricted" status, regardless of their permissions, which means `isFineGrained` should eventually be a `boolean` value. wdyt?

Reply:
So I think the function should look like:

```
export const hasEntitlement = (
  isFineGrained: boolean | null,
  entitlement: string,
  grantedEntitlements?: string[],
): boolean => {
  if (isFineGrained === false) {
    return true;
  }

  // if isFineGrained is null, we are still awaiting the api response to determine if the user has fine grained entitlements
  // in this case, we should not grant access to the user
  if (isFineGrained === null) {
    return false;
  }

  return (grantedEntitlements || []).includes(entitlement);
};
```

Reply:
> if (isFineGrained === null) {
>     return false;
>   }

I think we should return true in this case. In the end this decision is arbitrary, though I suspect most people accessing the UI to have server admin permissions.

Reply:
aah yes, logic was inverted here